src/layer/AbstractSentinelHubV3Layer.ts

```diff
@@ -132,6 +132,12 @@ export class AbstractSentinelHubV3Layer extends AbstractLayer {
         }
       }
 
+      //Convert internal evalscript to V3 if it's not in that version.
+      if (this.evalscript && !this.evalscript.startsWith('//VERSION=3')) {
+        let evalscriptV3 = await this.convertEvalscriptToV3(this.evalscript);
+        this.evalscript = evalscriptV3;
+      }
+
```

Comment:
Minor cleanup: since we are now converting to V3 here, can we maybe get rid of the lines 116-121?

Reply:
Lines 116-121 convert evalscripts given its url. So I think that they shouldn't be deleted.

Reply:
I meant, we should replace lines 115-122 with:
```
this.evalscript = response.data;
```

Then the code from line 135 on will take care of conversion (if needed). Or am I missing something?

Reply:
Yes of course, sorry I didn't get you.